evm/src/executor/inspector/cheatcodes/fork.rs

```diff
@@ -140,6 +143,59 @@ pub fn apply<DB: DatabaseExt>(
             )
             .map(empty)
             .map_err(Into::into),
+        HEVMCalls::GetLogs(inner) => {
+            // TODO: return error if active_fork_url is None
+            let url = data.db.active_fork_url()?;
+            if inner.0 > U256::from(u64::MAX) || inner.1 > U256::from(u64::MAX) {
+                return Some(Err(fmt_err!("Blocks in block range must be less than 2^64 - 1")));
+            }
+            if inner.3.len() > 4 {
+                return Some(Err(fmt_err!("Topics array must be less than 4 elements")));
+            }
+
+            // Taken from https://www.gakonst.com/ethers-rs/events/logs-and-filtering.html?highlight=logs#logs-and-filtering
+            // TODO: don't use `unwrap` below
+            let provider = ProviderBuilder::new(url).build().unwrap();
+            let mut filter = Filter::new()
+                .address(b160_to_h160(inner.2.into()))
+                .from_block(inner.0.as_u64())
+                .to_block(inner.1.as_u64());
+            for (i, item) in inner.3.iter().enumerate() {
+                match i {
+                    0 => filter = filter.topic0(U256::from(item)),
+                    1 => filter = filter.topic1(U256::from(item)),
+                    2 => filter = filter.topic2(U256::from(item)),
+                    3 => filter = filter.topic3(U256::from(item)),
+                    _ => unreachable!(),
```

Comment:
Maybe instead of panicking (which is fair in this case, as it should be impossible), we can also `eyre::bail` with an explicit error as to why this _should_ be impossible.